components/brave_perf_predictor/browser/third_party_extractor_unittest.cc

```diff
@@ -0,0 +1,116 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_perf_predictor/browser/third_party_extractor.h"
+
+#include <fstream>
+#include <streambuf>
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_perf_predictor {
+
+const char test_mapping[] = R"(
+[
+{
+    "name":"Google Analytics",
+    "company":"Google",
+    "homepage":"https://www.google.com/analytics/analytics/",
+    "categories":["analytics"],
+    "domains":["www.google-analytics.com","ssl.google-analytics.com","google-analytics.com","urchin.com"]
+},
+{
+    "name":"Facebook",
+    "homepage":"https://www.facebook.com",
+    "categories":["social"],
+    "domains":["www.facebook.com","connect.facebook.net","staticxx.facebook.com","static.xx.fbcdn.net","m.facebook.com","atlassbx.com","fbcdn-photos-e-a.akamaihd.net","23.62.3.183","akamai.net","akamaiedge.net","akamaitechnologies.com","akamaitechnologies.fr","akamaized.net","edgefcs.net","edgekey.net","edgesuite.net","srip.net","cquotient.com","demandware.net","platform-lookaside.fbsbx.com"]
+}
+])";
+
+class ThirdPartyExtractorTest : public ::testing::Test {
```

Comment:
you don't need to write an extra class, just use `TEST` instead of `TEST_F`. `LoadFile` could be a free function in an anonymous namespace. Maybe also rename it to `LoadDataset`?